d3a_interface/matching_algorithms/pay_as_clear_matching_algorithm.py

```diff
@@ -119,48 +118,46 @@ def _clearing_point_from_supply_demand_curve(bids_rate_energy: Dict, offers_rate
             if len(clearing) > 0:
                 return clearing[-1].rate, clearing[-1].energy
 
-    def get_clearing_point(self, bids: List[Dict], offers: List[Dict], current_time):
+    def get_clearing_point(self, bids: List[Dict], offers: List[Dict], current_time, market_id):
         self.sorted_bids = sort_list_of_dicts_by_attribute(bids, "energy_rate", True)
         self.sorted_offers = sort_list_of_dicts_by_attribute(offers, "energy_rate")
-        clearing = None
+        clearing, cumulative_bids, cumulative_offers = None, None, None
 
         if len(self.sorted_bids) == 0 or len(self.sorted_offers) == 0:
             return
 
         if ConstSettings.IAASettings.PAY_AS_CLEAR_AGGREGATION_ALGORITHM == 1:
             cumulative_bids = self._accumulated_energy_per_rate(self.sorted_bids)
             cumulative_offers = self._accumulated_energy_per_rate(self.sorted_offers)
-            self.state.cumulative_bids = cumulative_bids
-            self.state.cumulative_offers = cumulative_offers
```

Comment:
To repeat the offline discussion we had yesterday: Why not using the `self.state` as it was used before. With this you can omit passing the  `cumulative_bids` and `cumulative_offers` to `_get_clearing_point` and access them via `self.state` in there.
Or am I missing something here? If this needs to be done the way you implemented, please remove the self.state from the class because I dont see it being used anywhere else, right?

Reply:
`_get_clearing_point ` would be triggered multiple times within a slot, and not everytime there would be a clearing but still saving uncleared `cumulative_bids /cumulative_offers ` to state would lead to increase in memory which is also not really needed for the supply/demand curve. Regarding removing self.state from the class, is simply an organisation of markets clearing related data like, clearing rate, clearing energy, supply curve and demand cruve; and these data used for plotting the backend supply/demand curve.

Reply:
Ok, I guess I was confused by the fact that the `cumulative_bids`, `cumulative_offers` where passed around in the methods in this class while there should be also in the `state` member. I would have reused this one, but this is probably a matter of taste.